torch/lib/THD/base/data_channels/DataChannelGloo.cpp

```diff
@@ -0,0 +1,162 @@
+#include "DataChannelGloo.hpp"
+#include "DataChannelUtils.hpp"
+#include "Store.hpp"
+
+#include "gloo/allreduce_ring.h"
+#include "gloo/transport/tcp/device.h"
+#include "gloo/rendezvous/redis_store.h"
+#include "gloo/rendezvous/prefix_store.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+
+
+namespace thd {
+
+using ::gloo::rendezvous::Store;
+using ::gloo::rendezvous::PrefixStore;
+using ::gloo::rendezvous::Context;
+
+DataChannelGloo::DataChannelGloo()
+  : _rank(load_rank_env())
+  , _num_processes(load_world_size_env())
```

Comment:
Not sure if we really want this because that would mean we have to set this variable for workers too. I think we can use `Store` to handle this issue. Once we setup the store (as below) we can do something like:
```
if (rank == 0) {
  _num_processes = load_world_size_env();
  _store.set("world_size", std::to_string(_num_processes));
} else {
  auto world_size = _store.get("world_size");
  _num_processes = std::atoll(world_size.c_str());
}
```